### PR TITLE
Add guzzle

### PR DESCRIPTION
**Pull request by LukasReschke**
Another try…

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -80,3 +80,35 @@ ircmaxell/random-lib/composer.lock
...
 icewind/streams/tests
+
+guzzlehttp/guzzle/build/
+guzzlehttp/guzzle/docs/
+guzzlehttp/guzzle/tests/
+guzzlehttp/guzzle/.gitignore
+guzzlehttp/guzzle/.travis.yml
+guzzlehttp/guzzle/CHANGELOG.md
+guzzlehttp/guzzle/composer.json
+guzzlehttp/guzzle/LICENSE
```
Comment (MorrisJobke) [inline .gitignore]: I thought that we need to ship the license file?

File: .gitignore [hunk 2]
```diff
@@ -80,3 +80,35 @@ ircmaxell/random-lib/composer.lock
...
+guzzlehttp/ringphp/phpunit.xml.dist
+guzzlehttp/streams/tests/
+guzzlehttp/streams/.gitignore
+guzzlehttp/streams/.travis.yml
+guzzlehttp/streams/CHANGELOG.rst
+guzzlehttp/streams/LICENSE
+guzzlehttp/streams/Makefile
+guzzlehttp/streams/README.rst
+guzzlehttp/streams/composer.json
+guzzlehttp/streams/phpunit.xml.dist
```
Comment (MorrisJobke) [inline .gitignore]: maybe add following general rules:

```
.travis.yml
README*
tests
docs
Makefile
phpunit.xml.dist
```

This should shrink this a lot and makes it a lot more maintainable ;)

File: .gitignore (hunk 1 above)
Comment (LukasReschke) [inline .gitignore]: Last time I got told to remove it as well :see_no_evil: – Well… Then let's add it again here.

File: .gitignore (hunk 2 above)
Comment (LukasReschke) [inline .gitignore]: Can you create a dedicated issue? – This feels out-of-scope for me.
(as in: "I really don't know what third-party apps we ship that abuse the name `docs` for something else.")

File: .gitignore (hunk 1 above)
Comment (MorrisJobke) [inline .gitignore]: I have no idea, but that is just a feeling (because the files itself doesn't contain a license header)

File: .gitignore (hunk 2 above)
Comment (MorrisJobke) [inline .gitignore]: Let's move this to the "once we move composer into core"

File: .gitignore (hunk 1 above)
Comment (LukasReschke) [inline .gitignore]: *sigh* – yeah - will add once Jenkins is somewhat happy with https://github.com/owncloud/core/pull/14913

**Pull request merged**